drivers/video/video_mcux_csi.c

```diff
@@ -121,7 +121,16 @@ static int video_mcux_csi_set_fmt(const struct device *dev, enum video_endpoint_
 	unsigned int bpp = video_pix_fmt_bpp(fmt->pixelformat);
 	status_t ret;
 
-	if (!bpp || ep != VIDEO_EP_OUT) {
+	switch (ep) {
```

Comment:
That repeated pattern would probably deserve a macro or function.

Reply:
I will propose something in another PR when there will be a pretext to touch to the video drivers again (to limit review burden), but it seems possible to use macros with the new scheme:

```c
#define VIDEO_EP_OUT_SELECTED(ep, match) \
        ((ep) == VIDEO_EP_OUT || (ep) == VIDEO_EP_ALL || (ep) == (match))

#define VIDEO_EP_IN_SELECTED(ep, match) \
        ((ep) == VIDEO_EP_IN || (ep) == VIDEO_EP_ALL || (ep) == (match))

/* Filter away, 0x00 used when only 1 endpoint supported */

if (bpp == 0 || !VIDEO_EP_OUT_SELECTED(ep, 0x00)) {
       return -EINVAL;
}

/* Matches endpoint 0x00 or 0x01 or 0x02 */

if (VIDEO_EP_OUT_SELECTED(ep, 0x00)) {
       work_on_ep0();
}

if (VIDEO_EP_OUT_SELECTED(ep, 0x01)) {
       work_on_ep1();
}

if (VIDEO_EP_IN_SELECTED(ep, 0x02)) {
       work_on_ep2();
}

/* If ep == VIDEO_EP_ALL, then all endpoint are enacted... this works */
```